Lab1

YuriyLav opened the pull request.

YuriyLav committed:
Add files via upload

YuriyLav committed:
Delete hello.php

YuriyLav committed:
Create README.md

YuriyLav committed:
Update composer.json

YuriyLav committed:
Update index.php

YuriyLav committed:
Update Comment.php

YuriyLav committed:
Update User.php

mtyrtov commented on u8201549-lab1/composer.json:
```diff
@@ -0,0 +1,14 @@
+        "roave/security-advisories": "dev-latest"
+    },
+    "autoload": {
+        "classmap": [
```
Автозагрузка должна быть по PSR-4

mtyrtov commented on u8201549-lab1/classes/User.php:
```diff
@@ -0,0 +1,73 @@
+        );
+    }
+
+    public function getId() : int
```
Лишний пробел перед двоеточием

mtyrtov commented on u8201549-lab1/classes/User.php:
```diff
@@ -0,0 +1,73 @@
+        return $this->dateCreate;
+    }
+
+    public function printUser()
```
Не указан возвращаемый тип

mtyrtov commented on u8201549-lab1/classes/Comment.php:
```diff
@@ -0,0 +1,26 @@
+        return $this->message;
+    }
+
+    function info(): void
```
1. Не указана область видимости 
2. В идеале этот класс не должен заниматься выводом данных на экран (SOLID)

mtyrtov commented on u8201549-lab1/index.php:
```diff
@@ -0,0 +1,60 @@
+    ->addMethodMapping('loadValidatorMetadata')
+    ->getValidator();
+
+$user1= new User(16786776, 'Egor', 'egor@mail.ru', '43565657ghg');
```
Знаки равно прыгают

mtyrtov commented on u8201549-lab1/index.php:
```diff
@@ -0,0 +1,60 @@
+$dateTime = readline("Enter date in format 'd-m-Y H:i:s': ");
+
+$dateValidator = Validation::createValidator();
+$errors= $dateValidator->validate($dateTime,
```
```
$errors = $dateValidator->validate($dateTime, [
    new Assert\DateTime(), 
    new Assert\NotBlank(),
]);
```

mtyrtov commented on u8201549-lab1/index.php:
```diff
@@ -0,0 +1,60 @@
+[new Assert\DateTime(),
+new Assert\NotBlank(),
+]);
+    if(!empty($error)){
```
Табуляция

YuriyLav committed:
modified:   u8201549-lab1/classes/Comment.php
	modified:   u8201549-lab1/classes/User.php
	modified:   u8201549-lab1/composer.json
	modified:   u8201549-lab1/composer.lock
	modified:   u8201549-lab1/index.php
